libs/common/src/middleware/validate-mongoId/validate-mongoId.ts

```diff
@@ -0,0 +1,20 @@
+import { ObjectId } from 'mongodb';
+import { Request, Response, NextFunction } from 'express';
+import { HttpStatusCode } from 'axios';
+import { API_ERROR_CODE } from '../../constants/api-error-code';
+
+/**
+ * Validator for a path parameter which is a mongo id
+ * @param {string} paramName
+ * @returns {import('express').RequestHandler}
+ */
+export const validateMongoId = (paramName: string) => (req: Request, res: Response, next: NextFunction) => {
+  const pathParam = req.params[paramName] as string;
```

Comment:
I think this needs to be validated properly to confirm its not undefined (zod will probably allow it to be passed in as undefined though if you leave it)
```suggestion
  const pathParam = req.params[paramName];
```